src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,48 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    private final MachineProducer<T> truckProducer;
+    private final MachineProducer<T> bulldozerProducer;
+    private final MachineProducer<T> excavatorProducer;
+
+    public MachineServiceImpl() {
+        this.truckProducer = (MachineProducer<T>) new TruckProducer();
+        this.bulldozerProducer = (MachineProducer<T>) new BulldozerProducer();
+        this.excavatorProducer = (MachineProducer<T>) new ExcavatorProducer();
+    }
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines;
+        machines = type == Bulldozer.class ? bulldozerProducer.get()
+                : type == Excavator.class ? excavatorProducer.get()
+                : type == Truck.class ? truckProducer.get()
+                : null;
```

Comment:
its not the case when its better to use ternary operators, moreover, here it makes very hard to read code
replace with if-else or switch-case statements